### ISSUE TITLE
lib:Cabal-2 requires `directory >= 1.2.2`

### ISSUE THREAD

**Issue by hvr** [haskell/cabal#4606]
I got the following compile error on the 2.0 branch with `directory-1.2.1` (which happens to be the version included in GHC 7.8.4's global pkg db):

```
[150 of 168] Compiling Distribution.Simple.Test.LibV09 ( Distribution/Simple/Test/LibV09.hs, /stuff3/work/GitHub/hackage-server/dist-newstyle/build/x86_64-linux/ghc-7.8.4/Cabal-2.0.0.0/build/Distribution/Simple/Test/LibV09.o )

Distribution/Simple/Test/LibV09.hs:40:28:
    Module ‘System.Directory’ does not export ‘makeAbsolute’
```

the culprit appears to be e76c4820cfbefafabb377a887f8771d097e58587

**Comment by 23Skidoo** [comment]
Possible solutions:

* Use `canonicalizePath` (which throws an exception when the path refers to something that doesn't exist).
* Bump the lower bound of the dependency on `directory`.

**Comment by hvr** [comment]
Note that bumping the dependency on directory has consequences for the bootstrapping shell-script...

**Comment by 23Skidoo** [comment]
Yep, thx for reminding me. I'll need to check the versions in the bootstrap script before the release anyway.

**Comment by hvr** [comment]
@23Skidoo I tested whether we can compile lib:Cabal near lower-bounds (by faking the `makeAbsolute` function), by using GHC 7.4.2;

the resulting install-plan has the following direct dependencies for lib:Cabal,

```
  array-0.4.0.0
  base-4.5.1.0
  binary-0.5.1.0
  bytestring-0.9.2.1
  containers-0.4.2.1
  deepseq-1.3.0.0
  directory-1.1.0.2
  filepath-1.3.0.0
  ghc-prim-0.2.0.0
  old-time-1.1.0.0
  pretty-1.1.1.0
  process-1.1.0.1
  time-1.4
  unix-2.5.1.1
```

which would imply tweaking the lower bounds like so

```diff

diff --git a/Cabal/Cabal.cabal b/Cabal/Cabal.cabal
index 434ce62..efdc0c8 100644
--- a/Cabal/Cabal.cabal
+++ b/Cabal/Cabal.cabal
@@ -76,13 +76,13 @@ flag parsec-struct-diff
 
 library
   build-depends:
-    array      >= 0.1 && < 0.6,
+    array      >= 0.4 && < 0.6,
     base       >= 4.5 && < 5,
-    bytestring >= 0.9 && < 1,
-    containers >= 0.4 && < 0.6,
+    bytestring >= 0.9.2 && < 1,
+    containers >= 0.4.2 && < 0.6,
     deepseq    >= 1.3 && < 1.5,
     filepath   >= 1.3 && < 1.5,
-    pretty     >= 1.1 && < 1.2,
+    pretty     >= 1.1.1 && < 1.2,
     process    >= 1.1.0.1 && < 1.7,
     time       >= 1.4 && < 1.9
 
@@ -102,13 +102,12 @@ library
   if impl(ghc < 7.6)
     build-depends: ghc-prim >= 0.2 && < 0.3
 
-  if !os(windows)
-    build-depends:
-      unix >= 2.5 && < 2.8
-
   if os(windows)
     build-depends:
       Win32 >= 2.2 && < 2.6
+  else
+    build-depends:
+      unix >= 2.5.1 && < 2.8
 
   ghc-options: -Wall -fno-ignore-asserts -fwarn-tabs
   if impl(ghc >= 8.0)
```

**Comment by 23Skidoo** [comment]
Fixed in 560903cf8ff3e8e02532792353cbf19f2a55334f.
